lib/access-context.js

```diff
@@ -32,9 +32,12 @@ function AccessContext(context) {
   }
   context = context || {};
 
+  assert(context.registry,
+    'Application registry is mandatory in AccessContext but missing in provided context');
```

Comment:
Should we fall back to `loopback.registry` when `context.registry` is not provided, in order to preserve backwards compatibility? `loopback.registry` is the global singleton where models like `loopback.User` are defined.